torch_geometric/nn/conv/pna_conv.py

```diff
@@ -192,17 +193,16 @@ def __repr__(self):
                 f'edge_dim={self.edge_dim})')
 
     @staticmethod
-    def get_degree_histogram(loader) -> Tensor:
-        max_degree = 0
+    def get_degree_histogram(loader: DataLoader) -> Tensor:
+        deg_histogram = torch.zeros(1, dtype=torch.long)
         for data in loader:
             d = degree(data.edge_index[1], num_nodes=data.num_nodes,
                        dtype=torch.long)
-            max_degree = max(max_degree, int(d.max()))
-        # Compute the in-degree histogram tensor
-        deg_histogram = torch.zeros(max_degree + 1, dtype=torch.long)
-        for data in loader:
-            d = degree(data.edge_index[1], num_nodes=data.num_nodes,
-                       dtype=torch.long)
-            deg_histogram += torch.bincount(d, minlength=deg_histogram.numel())
+            d_bincount = torch.bincount(d, minlength=deg_histogram.numel())
```

Comment:
Awesome! Like it. We can also split this fix in a separate PR.

Reply:
I see, I will split the PR.